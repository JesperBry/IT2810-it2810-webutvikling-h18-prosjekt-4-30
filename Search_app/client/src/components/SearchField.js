import React from "react";
import { Field, reduxForm } from "redux-form";
import { connect } from "react-redux";
import { fetchProducts } from "../actions/productActions";

import "../styles/SearchField.css";

class SearchField extends React.Component {
  // Render a form Field from the redux-form lib
  render() {
    return (
      <div className="head-bar">
        <h1 className="head-title">POLUTFORSKER</h1>
        <form className="search-form" onSubmit={this.handleSubmit}>
          <Field
            name="search"
            component="input"
            type="text"
            placeholder="Search..."
          />
        </form>
      </div>
    );
  }

  // Handle the submit on the form
  // Providing a handler as a prop
  // Called by onSubmit in the redux-form
  handleSubmit = e => {
    e.preventDefault();
    this.props.search();
    this.props.page();
  };
}

// Receives the dispatch() method and return callback props.
const mapDispatchToProps = dispatch => ({
  search: () => dispatch(fetchProducts()),
  page: () => dispatch({ type: "PAGE_RESET" })
});

// Connects component to store and dispatch store state to the component props
SearchField = connect(
  () => ({}),
  mapDispatchToProps
)(SearchField);

export default reduxForm({ form: "searchValue" })(SearchField);
