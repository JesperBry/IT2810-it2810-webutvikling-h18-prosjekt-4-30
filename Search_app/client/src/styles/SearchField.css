@import url("https://fonts.googleapis.com/css?family=Roboto:300,400,500,700");

.head-title {
  font-family: "Roboto", sans-serif;
  font-weight: 300;
  font-size: 3rem;
  letter-spacing: 0.4rem;
  color: #ecf0f1;
  margin-bottom: 2rem;
}

.head-bar {
  display: flex;
  flex-direction: column;
  justify-content: center;
  background-color: #343a40;
  height: 15rem;
}

input::placeholder {
  color: #343a40;
}

input:focus {
  outline: none;
}

.search-form input[type="text"] {
  font-family: "Roboto", sans-serif;
  background-color: #f4f6f7;
  padding: 0.5rem;
  padding-left: 1.5rem;
  width: 38rem;
  margin-bottom: 2rem;
  border: none;
  font-size: 1.2rem;
  border-radius: 0.3rem;
}
