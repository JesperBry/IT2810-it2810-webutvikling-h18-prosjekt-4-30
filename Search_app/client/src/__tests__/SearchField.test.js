import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Enzyme, { shallow } from "enzyme";

import SearchField from "../components/SearchField";

it("Matches snapshot", () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<SearchField />);
  expect(result).toMatchSnapshot();
});
