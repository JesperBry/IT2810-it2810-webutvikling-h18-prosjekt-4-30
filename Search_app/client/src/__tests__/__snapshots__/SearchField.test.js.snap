// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<Connect(Form(Connect(SearchField)))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="searchValue"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
